packages/@aws-cdk/aws-iotevents/lib/state.ts

```diff
@@ -28,21 +28,46 @@ export class State {
    */
   public readonly stateName: string;
 
+  private transitionEvents: ITransitionEvent[] = []
+
   constructor(private readonly props: StateProps) {
     this.stateName = props.stateName;
   }
 
   /**
-   * Return the state property JSON
+   * Add a transition event to the state.
+   *
+   * @param transitionEvent the transition event that triggered if condition is evaluated to TRUE
+   */
+  public transitionTo(transitionEvent:ITransitionEvent) {
```

Comment:
I actually don't like the API of this method right now.

1. I think the state we are transitioning to should be the first positional argument, not in `ITransitionEvent`.
2. I think `eventName` should be optional, and we should generate an event name by default by combining the names of the States.
3. We call this second argument that's an object `Options` by convention in the CDK, so this should be `TransitionOptions`.

Reply:
So smart! 🤩 Thank you!